src/main/groovy/com/avast/gradle/dockercompose/ComposeSettings.groovy

```diff
@@ -117,6 +119,16 @@ class ComposeSettings {
     }
 
     void isRequiredBy(Task task) {
+        if (inheritInputAndOutputs) {
+            upTask.inputs.files task.inputs.files
+            downTask.inputs.files task.inputs.files
+            upTask.outputs.upToDateWhen {
```

Comment:
I'm not sure if I understand this line - according to the [docs](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskOutputs.html) the lambda passed to `upToDateWhen` should return a `bool` so this always returns _true value_ IMHO.

Reply:
You are right. My bad.
```
upTask.outputs.files task.outputs.files
downTask.outputs.files task.outputs.files
```
This works but uses deprecated API.

Reply:
It would IMHO work only in some cases, e.g. when a task has another types of inputs aside from `files`.

In fact, we want what is described here:
https://discuss.gradle.org/t/how-to-check-if-a-future-task-is-up-to-date/15189/6
https://stackoverflow.com/questions/27993814/only-run-task-if-another-isnt-up-to-date-in-gradle

It doesn't seem that there is a nice solution to this issue so I would prefer not to continue on this PR, sorry :(

Reply:
@augi Those inputs can be passed as well.

I came up with this, but I can create another plugin to do this.
```
upTask.outputs.cacheIf {
    true
}
downTask.outputs.cacheIf {
    true
}
upTask.inputs.properties(task.properties)
downTask.inputs.properties(task.properties)
task.inputs.files.each { f ->
    if (f.exists()) {
        if (f.isDirectory()) {
            upTask.inputs.dir f
            downTask.inputs.dir f
        } else {
            upTask.inputs.file f
            downTask.inputs.file f
        }
    }
}
task.outputs.files.each { f ->
    if (f.exists()) {
        if (f.isDirectory()) {
            upTask.outputs.dir f
            downTask.outputs.dir f
        } else {
            upTask.outputs.file f
            downTask.outputs.file f
        }
    }
}
```

Reply:
Can you release the yesterday's changes? Thanks